contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,265 @@
+
+    // TODO: has to redeem before calculating assets!!!
+    /// @notice this also includes shares from the pending withdrawal
+    function investedAssets()
+        public
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        IRyskLiquidityPool.DepositReceipt memory depositReceipt = ryskLqPool
+            .depositReceipts(address(this));
+        uint256 currentPricePerShare = _getWithdrawalPricePerShare();
+
+        // shares yet to be minted for pending deposits
+        uint256 pendingSharesAmount = 0;
+        if (depositReceipt.epoch != 0) {
+            uint256 depositPricePerShare = ryskLqPool.depositEpochPricePerShare(
+                depositReceipt.epoch
+            );
+
+            pendingSharesAmount =
+                (((depositReceipt.amount * SHARES_CONVERSION_FACTOR) /
+                    depositPricePerShare) * currentPricePerShare) /
+                SHARES_CONVERSION_FACTOR;
+        }
+
+        // shares marked by initiated withdrawal (now owned by the pool itself)
+        uint256 withdrawalSharesAmount = (pendingWithdrawal.shares *
+            ryskLqPool.withdrawalEpochPricePerShare(pendingWithdrawal.epoch)) /
+            SHARES_CONVERSION_FACTOR;
+
+        // entitled = redeemed + unredeemed
+        uint256 entitledSharesAmount = ((depositReceipt.unredeemedShares +
+            ryskLqPool.balanceOf(address(this))) * currentPricePerShare) /
+            SHARES_CONVERSION_FACTOR;
+
+        return
+            pendingSharesAmount + withdrawalSharesAmount + entitledSharesAmount;
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+        ryskLqPool.deposit(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 _amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (_amount == 0) revert StrategyAmountZero();
+        if (_amount > investedAssets()) revert StrategyNotEnoughShares();
+
+        _initializePendingWithdrawal();
+
+        uint256 sharesToWithdraw = _underlyingToShares(_amount);
+        pendingWithdrawal.shares += uint128(sharesToWithdraw);
+
+        emit RyskWithdrawalInitiated(_amount);
+        ryskLqPool.initiateWithdraw(sharesToWithdraw);
+    }
+
+    /**
+     * Completes the pending withdrawal initiated in an earlier epoch.
+     *
+     * @notice Expected to be called by the backend (keeper bot) once the Rysk liquidity pool enters a new withdrawal epoch.
+     * The backend should subscribe to 'WithdrawalEpochExecuted' event on the Rysk LiquidityPool contract,
+     * and act by calling this function when the event is emitted to complete pending withdrawal.
+     */
+    function completeWithdrawal() external {
```

Comment:
I guess this is missing a `onlyManager`?

Reply:
`onlyManager` is used for calls from the vault, this should be called by the backend. I'll add the KEEPER role for this

Reply:
[fixed](https://github.com/lindy-labs/sc_solidity-contracts/pull/152/commits/5e5f7afc6b4206ade7e6bffd3b6751c9da71fa61)

Reply:
On the comment, the details about the backend are likely wrong. We will use the subgraph to keep track of the executions.